src/ecdh.rs

```diff
@@ -87,6 +89,11 @@ impl SharedSecret {
         debug_assert!(len <= self.data.len());
         self.len = len;
     }
+
+    /// Serializes the whole shared secret buffer as byte value.
+    pub(crate) fn serialize_secret(&self) -> [u8; BUF_SIZE] {
```

Comment:
This name and signature are confusing because it looks like it's expensive operation while it just does memcpy. I think for 256 bytes long data it's better to use a reference.

Reply:
I'm going to refactor and use `into_secret`.